### PR TITLE
Added startup configuration option

### PR DESCRIPTION
**Pull request by miccico**
##### SUMMARY
Currently all containers that are created using this module start at the same time at system boot. In order to be able to reflect service dependencies and spread initial startup load proxmox supports the configuration of a startup parameter. This pull request adds the option to configure startup behavior of lxc containers. It works well in conjunction with onboot and allows to set startup order, startup delay and shutdown delay.

Changelog Fragment
  - proxmox - adds ``startup`` parameters for configuring startup order, startup delay and shutdown delay.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
proxmox

##### ADDITIONAL INFORMATION
```
sample usage:
startup: 
      - order=10
      - up=50
      - down=20
    
```

### CODE REVIEW

File: plugins/modules/proxmox.py [hunk 1]
```diff
@@ -81,6 +81,14 @@
...
     elements: str
     version_added: 2.0.0
+  startup:
+    description:
+      - Specifies the startup order of the container.
+      - order=# - where # is a non-negative number defining the general startup order. Shutdown in done with reverse ordering.
+      - up=# - where # is in seconds, which specifies a delay to wait before the next VM is started.
+      - down=# - where # is in seconds, which specifies a delay to wait before the next VM is stopped.
+    type: list
+    elements: str
```
Comment (felixfontein) [inline plugins/modules/proxmox.py:91]:
```suggestion
    elements: str
    version_added: 8.5.0
```

File: plugins/modules/proxmox.py [hunk 2]
```diff
@@ -81,6 +81,14 @@
     type: list
     elements: str
     version_added: 2.0.0
+  startup:
+    description:
+      - Specifies the startup order of the container.
+      - order=# - where # is a non-negative number defining the general startup order. Shutdown in done with reverse ordering.
+      - up=# - where # is in seconds, which specifies a delay to wait before the next VM is started.
+      - down=# - where # is in seconds, which specifies a delay to wait before the next VM is stopped.
```
Comment (felixfontein) [inline plugins/modules/proxmox.py:89]:
```suggestion
      - Use C(order=#) where C(#) is a non-negative number to define the general startup order. Shutdown in done with reverse ordering.
      - Use C(up=#) where C(#) is in seconds, to specify a delay to wait before the next VM is started.
      - Use C(down=#) where C(#) is in seconds, to specify a delay to wait before the next VM is stopped.
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment). Thanks.

**Review by krauthosting — APPROVED** [review]
Yep yep, changelog fragments and but seems otherwise straight forward :+1:

**Comment by miccico** [comment]
Sorry guys - still a newbie on working with projects that have big infrastructures :)

File: changelogs/fragments/8038-proxmox-startup.yml [hunk 3]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - proxmox - adds ``startup`` parameters to configure startup order, startup delay and shutdown delay. (https://github.com/ansible-collections/community.general/pull/8038).
```
Comment (felixfontein) [inline changelogs/fragments/8038-proxmox-startup.yml:2]:
```suggestion
  - proxmox - adds ``startup`` parameters to configure startup order, startup delay and shutdown delay (https://github.com/ansible-collections/community.general/pull/8038).
```

**Review by felixfontein — COMMENTED** [review]
(removed)

**Comment by felixfontein** [comment]
@miccico thanks for your contribution!
@krauthosting thanks for reviewing!

**Pull request merged**